frontend/catalyst/compilation_pipelines.py

```diff
@@ -500,6 +504,15 @@ def __init__(self, fn, compile_options):
         self._jaxpr = None
         self._mlir = None
         self._llvmir = None
+        self.function_name = None
+        self.preferred_workspace_dir = None
+        self.stored_compiled_functions = {}
+
+        # Make the format of static_argnums easier to handle.
+        if self.compile_options.static_argnums is None:
+            self.compile_options.static_argnums = ()
+        elif isinstance(self.compile_options.static_argnums, int):
+            self.compile_options.static_argnums = (self.compile_options.static_argnums, )
```

Comment:
Thank you for pointing this out. I will explore which option is better for implementation.

Reply:
Would you mind replying what you went with? :)

Reply:
I moved the code into the ``qjit`` function. Since a dataclass has already handled ``__init__``, I think duplicating ``__init__`` just for ``static_argnums`` might make the future update troublesome.

I just realized we can also use ``__post_init__``. Do you think we should switch to it?

Reply:
Oh nice, hadn't realized this was a thing. Sounds perfect